dataland-frontend/tests/e2e/specs/quality-assurance/CheckQaProcedure.ts

```diff
@@ -65,15 +80,45 @@ function testSubmittedDatasetIsInReviewList(companyName: string): void {
   safeLogout();
   login(uploader_name, uploader_pw);
 
-  testDatasetPresent(companyName, "APPROVED");
+  testDatasetPresentWithCorrectStatus(companyName, "APPROVED");
+}
+
+/**
+ * Tests that the item was added and is visible on the QA list
+ * @param storedCompany the stored company uploading the dataset
+ * @param dataset the data meta information that wa suploaded
+ */
+function testSubmittedDatasetIsInReviewListAndRejected(
+  storedCompany: StoredCompany,
+  dataset: DataMetaInformation
+): void {
+  login(reviewer_name, reviewer_pw);
+  cy.visitAndCheckAppMount("/qualityassurance");
+
+  cy.intercept(`**/api/metadata/${dataset.dataId}`).as("getMetadata");
+  cy.intercept(`**/api/companies/${storedCompany.companyId}`).as("getCompanyInformation");
+
+  cy.wait("@getMetadata").wait("@getCompanyInformation");
+
+  cy.get('[data-test="qa-review-section"] .p-datatable-tbody').last().click();
+  cy.get(".p-dialog").get('button[id="reject-button"]').should("exist").click();
+
+  safeLogout();
+  login(uploader_name, uploader_pw);
+
+  testDatasetPresentWithCorrectStatus(storedCompany.companyInformation.companyName, "REJECTED");
+
+  cy.visitAndCheckAppMount(`/companies/${storedCompany.companyId}/frameworks/lksg/${dataset.dataId}`)
+    .get('[data-test="editDatasetButton"')
+    .should("exist");
```

Comment:
Maybe we should actualy click the edit button to see that it works as intended, otherwise we might run into the case that the edit button is shown, but the data is not correctly retrieved after clicking it.

Reply:
Was added, ok now